src/PlaywrightSharp.Firefox/FirefoxPage.cs

```diff
@@ -246,7 +250,13 @@ public Task SetCacheEnabledAsync(bool enabled)
 
         public Task SetRequestInterceptionAsync(bool enabled) => throw new NotImplementedException();
 
-        public Task AuthenticateAsync(Credentials credentials) => throw new NotImplementedException();
+        public Task AuthenticateAsync(Credentials credentials)
+            => _session.SendAsync(
+                new NetworkSetAuthCredentialsRequest
+                {
+                    Username = credentials?.Username,
+                    Password = credentials?.Password,
+                }, options: _authenticateJsonOptions);

```

Comment:
We don't need the `_authenticateJsonOptions` variable, we could create the instance here. It won't be that expensive and we won't have that in memory all the time.